py/webdriver/edge/webdriver.py

```diff
@@ -0,0 +1,203 @@
+
+        if log_path:
+            warnings.warn('use service_log_path instead of log_path',
+                          DeprecationWarning, stacklevel=2)
+            service_log_path = log_path
+
+        self.port = port
+        if self.port == 0:
+            self.port = utils.free_port()
+
+        if self.use_chromium:
+            self.service = Service(
+                executable_path,
+                port=port,
+                verbose=verbose,
+                service_args=service_args,
+                log_path=service_log_path)
+            self.service.start()
+        else:
+            self.edge_service = Service(
+                executable_path, 
+                port=self.port, 
+                verbose=verbose, 
+                log_path=service_log_path)
+            self.edge_service.start()
+
+        try:
+            if self.use_chromium:
+                RemoteWebDriver.__init__(
+                    self,
+                    command_executor=EdgeRemoteConnection(
+                        remote_server_addr=self.service.service_url,
+                        keep_alive=keep_alive),
+                    desired_capabilities=desired_capabilities)
+            else:
+                RemoteWebDriver.__init__(
+                    self,
+                    command_executor=RemoteConnection(
+                        remote_server_addr='http://localhost:%d' % self.port,
+                        resolve_ip=False,
+                        keep_alive=keep_alive),
+                    desired_capabilities=capabilities)
+        except Exception:
+            self.quit()
+            raise
+        self._is_remote = False
+
+    def launch_app(self, id):
+        """Launches Edge app specified by id."""
+        return self.execute("launchApp", {'id': id})
+
+    def get_network_conditions(self):
+        """
+        Gets Edge network emulation settings.
+
+        :Returns:
+            A dict. For example:
+
+            {'latency': 4, 'download_throughput': 2, 'upload_throughput': 2,
+            'offline': False}
+
+        """
+        return self.execute("getNetworkConditions")['value']
+
+    def set_network_conditions(self, **network_conditions):
+        """
+        Sets Edge network emulation settings.
+
+        :Args:
+         - network_conditions: A dict with conditions specification.
+
+        :Usage:
+            driver.set_network_conditions(
+                offline=False,
+                latency=5,  # additional latency (ms)
+                download_throughput=500 * 1024,  # maximal throughput
+                upload_throughput=500 * 1024)  # maximal throughput
+
+            Note: 'throughput' can be used to set both (for download and upload).
+        """
+        self.execute("setNetworkConditions", {
+            'network_conditions': network_conditions
+        })
+
+    def execute_cdp_cmd(self, cmd, cmd_args):
+        """
+        Execute Edge Devtools Protocol command and get returned result
```

Comment:
Nit: Rename this to "Chromium Devtools Protocol", since there isn't really a separate Edge Devtools Protocol. Also this aligns with the "cdp" in the method name.

Reply:
Also update other usages of Edge devtools protocol throughout this PR.